transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/network/HttpClientProvider.kt

```diff
@@ -10,44 +11,67 @@ import io.ktor.client.plugins.HttpTimeout
 import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
 import io.ktor.client.plugins.logging.LogLevel
 import io.ktor.client.plugins.logging.Logging
-import io.ktor.client.plugins.retry
 import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.http.HttpHeaders
 import io.ktor.http.HttpStatusCode
 import io.ktor.serialization.kotlinx.json.json
+import io.ktor.util.AttributeKey
 
 private const val TIMEOUT_IN_MS = 30000L
-private const val MAX_RETRIES_ON_SERVER_ERRORS = 3
-private const val DELAY_BETWEEN_RETRIES_IN_MILLISECONDS = 5000L
-private val RETRYABLE_STATUS_CODES = setOf(
-    HttpStatusCode.InternalServerError,
-    HttpStatusCode.BadGateway,
-    HttpStatusCode.ServiceUnavailable,
-    HttpStatusCode.GatewayTimeout,
-)
-
-internal fun defaultHttpClient(logging: Boolean = false): HttpClient =
+private val SkipRetry = AttributeKey<Boolean>("transport.retry.skip")
+
+internal fun HttpRequestBuilder.noRetry() {
+    attributes.put(SkipRetry, true)
+}
+
+internal fun defaultHttpClient(configuration: Configuration): HttpClient =
     HttpClient {
-        if (logging) {
+        if (configuration.logging) {
             install(Logging) {
-                this.logger = Log(logging, LogTag.HTTP_CLIENT).ktorLogger
+                logger = Log(configuration.logging, LogTag.HTTP_CLIENT).ktorLogger
                 level = LogLevel.INFO
             }
         }
-        install(ContentNegotiation) {
-            json(WebMessagingJson.json)
-        }
+        install(ContentNegotiation) { json(WebMessagingJson.json) }
         install(HttpCallValidator)
         install(HttpTimeout) {
             socketTimeoutMillis = TIMEOUT_IN_MS
             connectTimeoutMillis = TIMEOUT_IN_MS
         }
-        install(HttpRequestRetry)
-    }
+        install(HttpRequestRetry) {
+            if (!configuration.retryEnabled) {
+                maxRetries = 0
+                return@install
+            }
+
+            maxRetries = 3
```

Comment:
Why magic number here? We had the MAX_RETRIES_ON_SERVER_ERRORS for the max retries, but it was deleted.